addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrlfInjectionScanRule.java

```diff
@@ -128,19 +130,23 @@ public void scan(HttpMessage msg, String param, String value) {
     private boolean checkResult(HttpMessage msg, String param, String attack) {
         Matcher matcher = patternCookieTamper.matcher(msg.getResponseHeader().toString());
         if (matcher.find()) {
-            newAlert()
-                    .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                    .setParam(param)
-                    .setAttack(attack)
-                    .setEvidence(matcher.group())
-                    .setMessage(msg)
-                    .raise();
+            buildAlert(param, attack, matcher.group(), msg).raise();
             return true;
         }
 
         return false;
     }
 
+    private AlertBuilder buildAlert(String param, String attack, String evidence, HttpMessage msg) {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                .setParam(param)
+                // If the attack isn't java escape then it's truncated at the CRLF
```

Comment:
What/where is truncating it?

Reply:
🤷‍♂️ I just know that when I printed it while gathering details I was only getting "any".

Reply:
Printing to the console? I'd not have expected this to be needed.

Reply:
This one is still pending.

Reply:
I believe it impacted how the alert was built as well. Will test and report back.

Reply:
/le sigh

Now I can't recreate 🤦‍♂️

Dropped.